services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -0,0 +1,881 @@
+            protectedIslsInfo.each { protectedIsl ->
+                if (isl.id == protectedIsl.id) {
+                    assert isl.availableBandwidth - protectedIsl.availableBandwidth == flow.maximumBandwidth
+                }
+            }
+        }
+
+        when: "Break ISL on the protected path (bring port down) to init the recalculate procedure"
+        def islToBreakProtectedPath = protectedIsls[0]
+        northbound.portDown(islToBreakProtectedPath.dstSwitch.dpId, islToBreakProtectedPath.dstPort)
+
+        Wrappers.wait(WAIT_OFFSET + discoveryInterval) {
+            assert islUtils.getIslInfo(islToBreakProtectedPath).get().state == IslChangeType.FAILED
+        }
+
+        then: "Protected path is recalculated"
+        def newProtectedPath
+        Wrappers.wait(WAIT_OFFSET) {
+            newProtectedPath = pathHelper.convert(northbound.getFlowPath(flow.id).protectedPath)
+            newProtectedPath != currentProtectedPath
+        }
+        Wrappers.wait(WAIT_OFFSET) { assert northbound.getFlowStatus(flow.id).status == FlowState.UP }
+
+        and: "Current path is not changed"
+        currentPath == pathHelper.convert(northbound.getFlowPath(flow.id))
+
+        and: "Bandwidth is reserved for new protected path on involved isls"
+        def newProtectedIsls = pathHelper.getInvolvedIsls(newProtectedPath)
+        def allLinks = northbound.getAllLinks()
+        def newProtectedIslsInfo = newProtectedIsls.collect { islUtils.getIslInfo(allLinks, it).get() }
+
+        allIsls.each { isl ->
+            newProtectedIslsInfo.each { protectedIsl ->
+                if (isl.id == protectedIsl.id) {
+                    assert isl.availableBandwidth - protectedIsl.availableBandwidth == flow.maximumBandwidth
+                }
+            }
+        }
+
+        and: "Reservation is deleted on the broken isl"
+        def originInfoBrokenIsl = allIsls.find {
+            it.destination.switchId == islToBreakProtectedPath.dstSwitch.dpId &&
+                    it.destination.portNo == islToBreakProtectedPath.dstPort
+        }
+        def currentInfoBrokenIsl = islUtils.getIslInfo(allLinks, islToBreakProtectedPath).get()
+
+        originInfoBrokenIsl.availableBandwidth == currentInfoBrokenIsl.availableBandwidth
+
+        when: "Restore port status"
+        northbound.portUp(islToBreakProtectedPath.dstSwitch.dpId, islToBreakProtectedPath.dstPort)
+        Wrappers.wait(WAIT_OFFSET + discoveryInterval) {
+            assert islUtils.getIslInfo(islToBreakProtectedPath).get().state == IslChangeType.DISCOVERED
+        }
+
+        then: "Path is not recalculated again"
+        pathHelper.convert(northbound.getFlowPath(flow.id).protectedPath) == newProtectedPath
+
+        and: "Cleanup: revert system to original state"
+        flowHelper.deleteFlow(flow.id)
+        database.resetCosts()
+    }
+
+    @Unroll
+    def "Unable to create protected path on #flowDescription flow on the same path"() {
```

Comment:
Need to re-phrase this somehow.

Reply:
fixed
`"Unable to create #flowDescription flow with protected path on the same path"`